sns/parity_math.c

```diff
@@ -1835,22 +1841,45 @@ static void bufvec_buf(struct m0_bufvec *bvec, struct m0_buf *buf,
 	uint8_t		       *buf_data;
 	uint32_t		seg_size;
 
-	M0_ENTRY("bvec=%p, buf=%p, count=%u, dir=%s",
-		 bvec, buf, count, (dir?"TRUE":"FALSE"));
+	M0_ENTRY("bvec=%p, buf=%p, count=%u", bvec, buf, count);
 
 	for (j = 0, i = 0; j < count; ++j, i = 0) {
 		m0_bufvec_cursor_init(&cursor, &bvec[j]);
 		buf_data = (uint8_t *)buf[j].b_addr;
 		do {
 			seg_size = bvec[j].ov_vec.v_count[i];
-			if (dir)
-				memcpy(&buf_data[i * seg_size],
-				       m0_bufvec_cursor_addr(&cursor),
-				       seg_size);
-			else
-				memcpy(m0_bufvec_cursor_addr(&cursor),
-				       &buf_data[i * seg_size],
-				       seg_size);
+			memcpy(&buf_data[i * seg_size],
+				m0_bufvec_cursor_addr(&cursor),
+				seg_size);
+			++i;
+			step = m0_bufvec_cursor_step(&cursor);
+		} while (!m0_bufvec_cursor_move(&cursor, step));
+	}
+
+	M0_LEAVE();
+}
```

Comment:
static void bufvec2buf(struct m0_bufvec *bvec, struct m0_buf *buf)
{
	struct m0_bufvec_cursor cursor;
	m0_bcount_t		step;
	uint32_t		i;
	uint8_t		       *buf_data;
	uint32_t		seg_size;

	M0_ENTRY("bvec=%p, buf=%p", bvec, buf);

	m0_bufvec_cursor_init(&cursor, bvec);
	buf_data = (uint8_t *)buf->b_addr;
        i = 0;
	do {
		seg_size = bvec->ov_vec.v_count[i];
		memcpy(&buf_data[i * seg_size],
				m0_bufvec_cursor_addr(&cursor),
				seg_size);
		++i;
		step = m0_bufvec_cursor_step(&cursor);
	} while (!m0_bufvec_cursor_move(&cursor, step));

	M0_LEAVE();
}
```

and then copy array of m0_bufvec to array of m0_buf:

```
static void bufvec_to_buf_copy(struct m0_bufvec *bvecs, struct m0_buf *bufs,
			       uint32_t count)
{
	uint32_t		j;

	M0_ENTRY("bvec=%p, buf=%p, count=%u", bvec, buf, count);

	for (j = 0; j < count; ++j) {
                bufvec2buf(&bvecs[j], &bufs[j])
	}

	M0_LEAVE();
}
```

Reply:
Done